instance/models/openedx_instance.py

```diff
@@ -213,17 +214,34 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
             )
         return backend_map, backend_conf
 
-    def set_active_vm_dns_records(self):
+    def set_active_vm_dns_records(self, deactivate_appserver=False):
         """
         Set DNS A records for all active app servers.
         """
         self.logger.info("Setting DNS records for active app servers...")
-        for i, appserver in enumerate(self.get_active_appservers(), 1):
-            ip_addr = appserver.server.public_ip
-            if ip_addr:
+        with cache.lock('appserver_dns_record_update_{}'.format(self.ref.instance_id)):
+            for i, appserver in enumerate(self.get_active_appservers(), 1):
+                ip_addr = appserver.server.public_ip
+
+                if ip_addr:
+                    domain = "vm{index}.{base_domain}".format(index=i, base_domain=self.internal_lms_domain)
+                    gandi.api.set_dns_record(domain, type="A", value=ip_addr)
+
+            if deactivate_appserver:
+                active_appservers = self.get_active_appservers()
```

Comment:
@pkulkark, nit: if the result of the previous `self.get_active_appservers()` call was stored in a variable, this duplicate call can be avoided.

Reply:
Done :+1: